### PR TITLE
Adds rubocop and rubycritic

### PR DESCRIPTION
**Pull request by cfcortes**
Rubocop gen added to  enforce community coding practices

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "workbench.colorCustomizations": {
+    "activityBar.background": "#2D3010",
+    "titleBar.activeBackground": "#3F4317",
+    "titleBar.activeForeground": "#FAFBF2"
+  }
+}
```
Comment (EduardoGHdez) [inline .vscode/settings.json:7]: This is not part of the rubocop setup, and I wonder if we can get rid of it? 😄 

I mean, this are editor preferences, and aren't needed to being able to run this project 🙌🏽 

WDYT?

File: Gemfile [hunk 2]
```diff
@@ -5,3 +5,4 @@ gemspec
 
 gem "rake", "~> 12.0"
 gem "rspec", "~> 3.0"
+gem "rubocop", require: false
```
Comment (EduardoGHdez) [inline Gemfile]: We can group it to being only available in `dev` and `test` like this:
```ruby
gem "rubocop", group: [:test, :development]
```

WDYT?

File: Gemfile.lock [hunk 3]
```diff
@@ -45,6 +46,15 @@ GEM
...
       rainbow (>= 2.0, < 4.0)
+=======
+    ast (2.4.2)
+    diff-lcs (1.4.4)
+    parallel (1.20.1)
+    parser (3.0.2.0)
+      ast (~> 2.4.1)
+    rainbow (3.0.0)
+    rake (12.3.3)
+>>>>>>> 7cc27735a097b1874be8a8dc486074d5aba86860
```
Comment (EduardoGHdez) [inline Gemfile.lock]: Let's solve this git conflicts 😄 , otherwise, the `lock` file won't work

File: .vscode/settings.json (hunk 1 above)
Comment (cfcortes) [reply .vscode/settings.json:7]: Yes! added this file to .gitignore 
I made a commit for this as I was unsure how to add this update

File: Gemfile (hunk 2 above)
Comment (cfcortes) [reply Gemfile]: Yes! i did this same for rubycritic

File: Gemfile.lock (hunk 3 above)
Comment (cfcortes) [reply Gemfile.lock]: Removed file and run gem install

**Review by EduardoGHdez — COMMENTED** [review]
Great job here!

Do you mind squashing all commits into a single one? 😄 
Also, it'd be great if we can get rid of the vscode file

**Review by EduardoGHdez — APPROVED** [review]

**Pull request merged**